ingest/src/main/java/com/microsoft/azure/kusto/ingest/Validation.java

```diff
@@ -0,0 +1,48 @@
+package com.microsoft.azure.kusto.ingest;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class Validation {
+
+    public static void validateIsNotBlank(String str, String message) {
+        if (StringUtils.isBlank(str)) {
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    public static void validateIsNotNull(Object obj, String message) {
+        if (obj == null) {
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    public static void validateFileExists(File file, String message) throws FileNotFoundException {
+        validateIsNotNull(file, "file is null");
+
+        if (!file.exists()) {
+            throw new FileNotFoundException(message);
+        }
+    }
+
+    public static File validateAndCreateFile(String filePath) throws FileNotFoundException {
+        validateIsNotBlank(filePath, "filePath is blank");
+
+        File file = new File(filePath);
+        validateFileExists(file, "file does not exist: " + filePath);
+        return file;
+    }
+
+    public static URI validateAndCreateUri(String uri) {
```

Comment:
I don't think this method is needed. When one will try to create the uri, it will get the correct exception type URISyntaxException . We don't need to convert it to another exception.